File: src/test/kotlin/me/rajayudhaps/boilerplate/spi/CalculatorTests.kt
package me.rajayudhaps.boilerplate.spi

import me.rajayudhaps.boilerplate.factory.CalculatorFactory
import org.junit.jupiter.api.Assertions.assertNotNull
import org.junit.jupiter.api.BeforeAll
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.TestInstance
import kotlin.test.assertEquals

@TestInstance(TestInstance.Lifecycle.PER_CLASS)
class CalculatorTests {

  lateinit var calculator: Calculator

  @BeforeAll
  fun setup() {
    calculator = CalculatorFactory.getCalculator()
  }

  @Test
  fun hasInstance() {
    assertNotNull(calculator)
  }

  @Test
  fun performAddTest() {
    assertEquals(6, calculator.add(2, 4))
  }
}

File: calculator/src/test/clojure/me/rajayudhaps/boilerplate/calculator/calculator_tests.clj
(ns me.rajayudhaps.boilerplate.calculator.calculator-tests 
  (:require [clojure.test :as t]
            [me.rajayudhaps.boilerplate.calculator.calculator-impl :as spi]))

(t/deftest add-test
  (t/testing "add works as expected"
    (t/is (= 6 (spi/-add nil 2 4)))))